ci/lib/stage-test-examples.jenkinsfile

```diff
@@ -223,22 +223,34 @@ stage('test-examples') {
         }
     }
 
-    if (["ubuntu", "debian"].contains(env.os_release_id)) {
-       if (env.no_cpu.toInteger() > 16) {
-            try {
-                timeout(time:30,unit: 'MINUTES') {
-                  sh '''
+    if (env.no_cpu.toInteger() > 16) {
```

Comment:
There seems to be lots of common code that runs for Ubuntu/Cent, can you optimize this so that only the diff code is put inside the if blocks? For ex: you can assume the default as Ubuntu and have the if condition for Cent/RHEL.